packages/babel-parser/src/plugin-utils.js

```diff
@@ -1,19 +1,46 @@
 // @flow
 
 import type Parser from "./parser";
+import type { PluginConfig } from "./parser/base";
 
-export type Plugin = string | [string, Object];
+export type Plugin = PluginConfig;
 
-export type PluginList = $ReadOnlyArray<Plugin>;
+export type PluginList = $ReadOnlyArray<PluginConfig>;
 
 export type MixinPlugin = (superClass: Class<Parser>) => Class<Parser>;
 
-export function hasPlugin(plugins: PluginList, name: string): boolean {
-  return plugins.some(plugin => {
-    if (Array.isArray(plugin)) {
-      return plugin[0] === name;
+// This function’s second parameter accepts either a string (plugin name) or an
+// array pair (plugin name and options object). If an options object is given,
+// then each value is non-recursively checked for identity with the actual
+// option value of each plugin in the first argument (which is an array of
+// plugin names or array pairs).
+export function hasPlugin(
+  plugins: PluginList,
+  expectedConfig: PluginConfig,
+): boolean {
+  // The expectedOptions object is by default an empty object if the given
+  // expectedConfig argument does not give an options object (i.e., if it is a
+  // string).
+  const [expectedName, expectedOptions] =
+    typeof expectedConfig === "string" ? [expectedConfig, {}] : expectedConfig;
+
+  const expectedOptionsIsEmpty =
+    [...Object.entries(expectedOptions)].length === 0;
```

Comment:
`Object.entries` returns an array, no need to spread. Also, it's later used for looping, so perhaps worth saving.

Reply:
It looks like the Object.entries function isn’t available in Node 6 anyway. So we can’t use it...